blockmanager.go

```diff
@@ -3043,4 +3085,188 @@ func (l *lightHeaderCtx) RelativeAncestorCtx(
+				prevCheckpoint := b.findPreviousHeaderCheckpoint(
+					node.Height,
+				)
+
+				log.Infof("Rolling back to previous validated "+
+					"checkpoint at height %d/hash %s",
+					prevCheckpoint.Height,
+					prevCheckpoint.Hash)
+
+				err = b.rollBackToHeight(uint32(prevCheckpoint.Height))
+				if err != nil {
+					panic(fmt.Sprintf("Rollback failed: %s", err))
+					// Should we panic here?
+				}
+				tipHeader, height, err := b.cfg.BlockHeaders.ChainTip()
+				if err != nil {
+					return
+				}
+				b.headerList.ResetHeaderState(headerlist.Node{
+					Height: int32(height),
+					Header: *tipHeader,
+				})
+				return
+			}
+
+		}
+
+		// Verify checkpoint only if verification is enabled
+		if b.nextCheckpoint != nil && headerErr == nil && b.cfg.sideLoad.Verify &&
+			node.Height == b.nextCheckpoint.Height {
+
+			nodeHash := node.Header.BlockHash()
+			if nodeHash.IsEqual(b.nextCheckpoint.Hash) {
+
+				log.Infof("Verified downloaded block "+
+					"header against checkpoint at height "+
+					"%d/hash %s", node.Height, nodeHash)
+			} else {
+				log.Warnf("Error at checkpoint while side loading headers, exiting"+
+					"%d/hash %s", node.Height, nodeHash)
+				return
+			}
+
+		}
+
+		if header != nil {
+			headerBatch = append(headerBatch, headerfs.BlockHeader{
+				BlockHeader: header,
+				Height:      uint32(node.Height),
+			})
+
+		}
+
+		// Write header if batch is greater than the MaxHeaderWrite or if we have reached the
+		// end of file.
+		if len(headerBatch) >= MaxHeaderWrite || headerErr == io.EOF {
+			err = b.cfg.BlockHeaders.WriteHeaders(headerBatch...)
+			if err != nil {
+				log.Warnf("Error writing headers in blockheader store")
+				return
+			}
+			log.Debugf("Written header batch .... new tip %v", int(b.headerTip)+len(headerBatch))
+			// Updated header tip so as to enable cfhandler to start fetching cfheaders
+
+			if headerErr == io.EOF {
+				log.Infof("Successfully completed sideloading")
+				return
+			}
+			b.nextCheckpoint = b.findNextHeaderCheckpoint(node.Height)
+			b.newHeadersMtx.Lock()
+			b.headerTip = uint32(node.Height)
+			b.headerTipHash = node.Header.BlockHash()
+			b.newHeadersMtx.Unlock()
+			b.newHeadersSignal.Broadcast()
+
+			b.blkHeaderProgressLogger.LogBlockHeight(
+				header.Timestamp, node.Height,
+			)
+
+			headerBatch = make([]headerfs.BlockHeader, 0)
+
+		}
+
+		b.headerList.PushBack(*node)
+		// Check if we are to quit now
+		select {
+		case <-b.quit:
+			return
+		default:
+		}
+	}
+
+}
+
+func (b *blockManager) verifyBlockHeader(blockHeader *wire.BlockHeader, prevNode headerlist.Node) (error, bool) {
+	prevNodeHeader := prevNode.Header
+	prevHash := prevNode.Header.BlockHash()
+	prevNodeHeight := prevNode.Height
+
+	if prevHash.IsEqual(&blockHeader.PrevBlock) {
+		err := b.checkHeaderSanity(blockHeader, b.headerList, prevNodeHeight, &prevNodeHeader)
+
+		if err != nil {
+			return fmt.Errorf("did not pass sanity check: %v", err), true
+		}
+		return nil, true
+
+	} else {
```

Comment:
we can put this outside the `else`

Reply:
put what please?

Reply:
```
// verifyBlockHeader verifies blockheader by checking if it connects to the previous block and its block header sanity.
func (b *blockManager) verifyBlockHeader(blockHeader *wire.BlockHeader, prevNode headerlist.Node) (bool, error) {
	prevNodeHeader := prevNode.Header
	prevHash := prevNode.Header.BlockHash()
	prevNodeHeight := prevNode.Height

	if prevHash.IsEqual(&blockHeader.PrevBlock) {
		err := b.checkHeaderSanity(blockHeader, false, prevNodeHeight, &prevNodeHeader)

		if err != nil {
			return true, fmt.Errorf("did not pass sanity check: %w", err)
		}
		return true, nil

	}
	
	// no need for else, because the if has a return
	return false, nil
}
```

Reply:
Thanks